### PR TITLE
Review after PR 31: package-lock.json update and restructure readme

### PR DESCRIPTION
**Pull request by beccaelenzil**
This PR:
- update package-lock.json
- update skills assessed (from learning goals)
    - some skills assessed taken from [react tic tac toe](https://github.com/adagold/react-tic-tac-toe)
- moved project instructions from readme to project docs
- scaffold for wave 3 and wave 4

To do in future PRs that will branch from be/c17-update:
- Add more details to wave 1 and 2 (including prop)
- Write requirements for wave 3 and 4
- Update image (gif)
- Write tests for wave 3 and 4

### CODE REVIEW

File: project-docs/wave-01.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+A good way to test this code as you write it would be to take the first chat message from the JSON data file and use it as the data for the component(s) you are implementing.
+
+## Tests
+
+This component has a set of tests that ensure that props passed to the component appear in the browser.  To pass the tests you should name this component **ChatEntry** and give it props which match, (including the name of the props) the elements from each chat message in the JSON data file.  Otherwise the tests will not pass.  Alternatively you can adjust the tests to use the proper props.
+
+## Styling
+The styles necessary to make an app which visually matches the demo above have been provided for you, but you will need to make use of them in your React components by adding classes to specific HTML elements in your JSX.
+
+There are several CSS classes for the different elements of a chat message. You should attempt to determine which classes should be applied to which HTML elements, however if you are stuck on this please ask a member of the instructional staff as the styling on this project is not the learning goal.
```
Comment (kp483) [inline project-docs/wave-01.md:13]: Maybe clarify they can reach out for help beyond staff if they are not available. 

```suggestion
There are several CSS classes for the different elements of a chat message. You should attempt to determine which classes should be applied to which HTML elements yourself. However, if you are stuck on this, please reach out for help as the styling is not the primary learning goal of this project.
```

File: project-docs/wave-02.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Wave 2
+
+In this wave you will extend from displaying a single chat message to an entire chat log. The focus here should be on creating component(s) that can display a sequence of individual chat message components.
```
Comment (kp483) [inline project-docs/wave-02.md:3]: Nitpicky 
```suggestion
In this wave you will extend functionality from displaying a single chat message to displaying an entire chat log. The focus here should be on creating component(s) that can display a sequence of individual chat message components.
```

File: project-docs/optional-enhancements.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+## Local and Remote messages
+At this point you should have a chat log which shows chat messages from two different people, however all of the chat messages are displayed to one side of the screen. In keeping with the design standards for messenger apps however, one person's messages should be displayed to the other side of the screen.
+
+In order to do this you will need to figure out a way, without modifying the JSON file itself, to determine whether each message is "local" or "remote". There are different ways to decide that, and because this is not a real-world application you should feel free to pick whatever way suits you.
+
+Once you can designate each message as either local or remote, you should modify your chat message component(s) to include the CSS class "local" or "remote" as appropriate. This class should be added to whatever HTML element has the "chat-entry" class.
+
+## Header title
+One last finishing touch, the application title in the header. The text of the title should be determined dynamically based on the names of the participants in the chat log data. Consult the demo above for an example of what the title might contain.
```
Comment (kp483) [inline project-docs/optional-enhancements.md:11]: Nitpicking

```suggestion
One last finishing touch: the application title in the header. The text of the title should be determined dynamically based on the names of the participants in the chat log data. Consult the demo above for an example of what the title might contain.
```

**Review by kp483 — APPROVED** [review]
Looks good! I left some minor suggestions on wording in the project docs.

Didn't do a super close reading of package-lock.json since the differences were huge, but ran project setup and everything looked good.

File: project-docs/optional-enhancements.md (hunk 3 above)
Comment (beccaelenzil) [reply project-docs/optional-enhancements.md:11]: this content is outdated

**Pull request merged**